packages/react-accordion/src/components/AccordionHeader/useAccordionHeader.ts

```diff
@@ -59,53 +44,30 @@ export const useAccordionHeader = (
   const id = useId('accordion-header-', props.id);
   const panel = useDescendants(accordionItemDescendantContext)[1] as AccordionItemDescendant | undefined;
   const innerRef = React.useRef<HTMLElement>(null);
-  const state = mergeProps(
-    {
-      ref: useMergedRefs(ref, innerRef),
-      size: 'medium' as AccordionHeaderSize,
-      inline: false,
-      expandIcon: {
-        as: DefaultExpandIcon,
-        'aria-hidden': true,
-      },
-      button: {
-        id,
-        onClick: onAccordionHeaderClick,
-        'aria-disabled': disabled,
-        'aria-controls': panel?.id,
-      },
-      as: 'div',
-      role: 'heading',
-      expandIconPosition: 'start',
-      context: {
-        disabled: false,
-        open: false,
-        size: 'medium',
-        expandIconPosition: 'start',
-      },
+  const initialState = {
+    ...props,
+    ref: useMergedRefs(ref, innerRef),
+    size: props.size ?? size ?? 'medium',
+    inline: props.inline ?? inline ?? false,
+    role: props.role ?? 'heading',
+    expandIconPosition: props.expandIconPosition ?? expandIconPosition ?? 'start',
+    components: {
+      root: 'div',
+      button: 'button',
+      expandIcon: AccordionHeaderExpandIcon,
     },
-    resolveShorthandProps<AccordionHeaderProps, AccordionHeaderShorthandPropsCompat>(
-      { button, icon, expandIconPosition, expandIcon, size, inline },
-      accordionHeaderShorthandPropsCompat,
-    ),
-    defaultProps && resolveShorthandProps(defaultProps, accordionHeaderShorthandPropsCompat),
-    resolveShorthandProps(props, accordionHeaderShorthandPropsCompat),
-  );
-  const originalButtonKeyDown = state.button.onKeyDown;
-  state.button.onKeyDown = useEventCallback((ev: React.KeyboardEvent<HTMLElement>) => {
-    if (shouldPreventDefaultOnKeyDown(ev)) {
-      if (disabled) {
-        ev.preventDefault();
-        ev.stopPropagation();
-        return;
-      }
-      ev.preventDefault();
-      onAccordionHeaderClick(ev);
-    }
-    originalButtonKeyDown?.(ev);
-  });
-
-  useARIAButton(state.button);
+    icon: resolveShorthand(props.icon ?? icon),
+    expandIcon: resolveShorthand(props.expandIcon ?? expandIcon, {
+      'aria-hidden': true,
+    }),
+    button: useARIAButton(props.button ?? button, {
+      id,
+      onClick: onAccordionHeaderClick,
```

Comment:
This callback should be defined as an override to avoid collisions with user's input:

```tsx
state.button.onClick = () => {}
```